web_monitoring/internetarchive.py

```diff
@@ -90,162 +90,171 @@ def cdx_hash(content):
     return b32encode(hashlib.sha1(content).digest()).decode()
 
 
-def search_cdx(params, session=None):
+class CDXClient:
     """
-    Search archive.org's CDX API for all captures of a given URL.
-
-    This will automatically page through all results for a given search.
-
-    Returns an iterator of CdxRecord objects. The StopIteration value is the
-    total count of found captures.
-
-    Note that even URLs without wildcards may return results with different
-    URLs. Search results are matched by url_key, which is a SURT-formatted,
-    canonicalized URL:
-
-      * Does not differentiate between HTTP and HTTPS
-      * Is not case-sensitive
-      * Treats `www.` and `www*.` subdomains the same as no subdomain at all
+    A client for querying Wayback Machine's CDX API.
 
     Parameters
     ----------
-    params : dict
-        Any options that the CDX API takes. Must at least include `url`.
-    session : requests.Session, optional
-        A session object to use for requests.
-
-    Raises
-    ------
-    UnexpectedResponseFormat
-        If the CDX response was not parseable.
-
-    References
-    ----------
-    * https://github.com/internetarchive/wayback/tree/master/wayback-cdx-server
+    session : :class:`requests.Session`, optional
```

Comment:
@danielballan totally out of scope here, but this brought things back to mind… what do you think about adding type annotations? (I’ve never used them in Python) Can our doc tools leverage them?

Reply:
I have started using them in situation where I want extra code introspection, such as annotating a function's parameters so that I can automatically generate a GUI with the right input types (text box, select box, etc.). I'm not yet won over that they are worth the verbosity for their own sake on small- to medium-sized projects.